cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -54,6 +54,13 @@ func (rm *AzureResourceManager) GetDeploymentResourceOperations(ctx context.Cont
 		}
 	}
 
+	if resourceGroupName == "" {
+		for _, operation := range subOperations {
+			resourceGroupName = operation.Properties.TargetResource.ResourceGroup
+			break
+		}
+	}
```

Comment:
Is this addressing a new issue that occurs with resource group scopes? Is it better to have explicit conditional handling based on the type of scope? For example, if scope is resource group scope, I think the resource group name would be provided directly by the scope.

The fallback seems a little unintuitive unless we can codify in comments how it applies to the general case.